### PR TITLE
[INLONG-5577][Manager][Sort] Support function fieldType

### LINKED ISSUE #5577: [Feature][Manager][Sort] Support function fieldType
### Description

Support function fieldType

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by emhui**
### Prepare a Pull Request

[INLONG-5577][Manager][Sort] Support function fieldType

- Fixes #5577 

### Motivation

Support for defining fields as function types

### Modifications

When FieldType is set to a function type, there is no need to add single quotation marks

### CODE REVIEW

**Review by healchow — APPROVED** [review]

**Comment by gong** [comment]
Add a UT. How to use it.

**Review by LvJiancheng — APPROVED** [review]

**Comment by emhui** [comment]
> Add a UT. How to use it.

1

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/PulsarExtractNode.java [hunk 1]
```diff
@@ -44,7 +44,6 @@ public class PulsarExtractNode extends ExtractNode {
     @Nonnull
     @JsonProperty("topic")
     private String topic;
-    @Nonnull
```
Comment (healchow) [inline inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/PulsarExtractNode.java:47]: Why remove the validation in this PR?

**Review by healchow — CHANGES_REQUESTED** [review]

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/transformation/function/CustomFunction.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    /**
+     * Function param name
+     *
+     * @return The name of this function param
+     */
+    @Override
+    public String getName() {
+        return "CUSTOMFUNCTION";
```
Comment (yunqingmoswu) [inline inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/transformation/function/CustomFunction.java:58]: It is recommended to throw an exception directly here, and this method is not allowed to be accessed.

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/PulsarExtractNode.java (hunk 1 above)
Comment (emhui) [reply inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/PulsarExtractNode.java]: Has been corrected.

**Review by gong — APPROVED** [review]

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/transformation/function/CustomFunction.java (hunk 2 above)
Comment (emhui) [reply inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/transformation/function/CustomFunction.java]: Thank you， this is a good suggestion.

**Review by yunqingmoswu — APPROVED** [review]
LGTM

**Review by healchow — APPROVED** [review]

**Review by aloyszhang — APPROVED** [review]

**Pull request merged**